src/battle/overlay_12_0224E4FC.c

```diff
@@ -2883,3 +2884,414 @@ BOOL CheckMoveCallsOtherMove(u16 moveNo) {
     }
     return FALSE;
 }
+
+BOOL CurseUserIsGhost(BATTLECONTEXT *ctx, u16 moveNo, int battlerId) {
+    return (moveNo == MOVE_CURSE && (GetBattlerVar(ctx, battlerId, BMON_DATA_TYPE_1, NULL) == TYPE_GHOST || GetBattlerVar(ctx, battlerId, BMON_DATA_TYPE_2, NULL) == TYPE_GHOST));
+}
+
+BOOL CanStealHeldItem(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId) {
+    BOOL ret = FALSE;
+    int side = BattleSystem_GetFieldSide(bsys, battlerId);
+    
+    if (ctx->battleMons[battlerId].item && !(ctx->fieldSideConditionData[side].battlerBitKnockedOffItem & MaskOfFlagNo(ctx->selectedMonIndex[battlerId])) && !ItemIdIsMail(ctx->battleMons[battlerId].item)) {
+        ret = TRUE;
+    }
+    
+    return ret;
+}
+
+BOOL CanTrickHeldItem(BATTLECONTEXT *ctx, int battlerId) {
+    return !ItemIdIsMail(ctx->battleMons[battlerId].item);
+}
+
+BOOL WhirlwindCheck(BattleSystem *bsys, BATTLECONTEXT *ctx) {
+    BOOL ret = FALSE;
+    
+    if (ctx->battleMons[ctx->battlerIdAttacker].level >= ctx->battleMons[ctx->battlerIdTarget].level) {
+        ret = TRUE;
+    } else {
+        int level = ((BattleSystem_Random(bsys) & 0xFF) * (ctx->battleMons[ctx->battlerIdAttacker].level + ctx->battleMons[ctx->battlerIdTarget].level) >> 8) + 1;
+        
+        if (level > ctx->battleMons[ctx->battlerIdTarget].level / 4) {
+            ret = TRUE;
+        }
+    }
+    
+    return ret;
+}
+
+u8 GetBattlerAbility(BATTLECONTEXT *ctx, int battlerId) {
+    if ((ctx->battleMons[battlerId].moveEffectFlags & MOVE_EFFECT_GASTRO_ACID) && ctx->battleMons[battlerId].ability != ABILITY_MULTITYPE) {
+        return 0;
+    } else if ((ctx->fieldCondition & FIELD_CONDITION_GRAVITY) && ctx->battleMons[battlerId].ability == ABILITY_LEVITATE) {
+        return 0;
+    } else if ((ctx->battleMons[battlerId].moveEffectFlags & MOVE_EFFECT_INGRAIN) && ctx->battleMons[battlerId].ability == ABILITY_LEVITATE) {
+        return 0;
```

Comment:
do we want a constant for invalid/no ability?

Reply:
just checked and we already have one, will put the const here